Convert breaking change template to a form (#469)
aspnet/Announcements: .github/ISSUE_TEMPLATE/breaking-change.md

name: "ASP.NET Core breaking change"

> description: Report a change in ASP.NET Core that breaks something that worked in a previous version. Intended mostly for product-team use.

Reviewer: ```suggestion
description: Report a change in ASP.NET Core that breaks something that worked in a previous version. This is Intended to be used by the product-team.
```

title: "[Breaking change]: "
labels: "Breaking change"
assignees:

>  - gewarren

Reviewer: Any specific reason for you to be auto-assigned to these announcements?

Author: So I get notified to document it. I can leave it out if you prefer.

body:
  - type: textarea
    id: description
    attributes:
      label: Description
      description: Brief description of the breaking change.
    validations:
      required: true
  - type: dropdown
    id: version
    attributes:
      label: Version
      description: What version of .NET introduced the breaking change?
      options:
        - .NET 6 Preview 6
        - .NET 6 Preview 7
        - .NET 6 RC1
        - .NET 6 RC2

>         - .NET 6 GA

Reviewer: Should these only include RC1, RC2 and GA? I assume most of the preview7 breaking changes (if any) are already announced)

    validations:
      required: true
  - type: textarea
    id: new-behavior
    attributes:
      label: New behavior
      description: Describe the new behavior. Include code snippets if applicable.
    validations:
      required: true
  - type: checkboxes
    id: change-type
    attributes:
      label: Type of breaking change
      options:
        - label: Binary incompatible (code must be recompiled to use the newer API version)
        - label: Source incompatible (code must be changed and recompiled to use the newer API version)
    validations:
      required: true
  - type: textarea
    id: reason
    attributes:
      label: Reason for change
      description: Describe why the breaking change was introduced.
    validations:
      required: true
  - type: textarea
    id: recommended-action
    attributes:
      label: Recommended action
      description: Describe the recommended action an affected user should take, such as workarounds or examples of code changes.
    validations:
      required: true
  - type: dropdown
    id: area
    attributes:
      label: Feature area
      description: What feature area(s) does the change affect?
      options:

>         - ASP.NET Core (Default)

Reviewer: Is this field necessary at all, given that there is only this one option?

Author: Probably not, I can remove it.
